### DISCUSSION TITLE
Objects appear as Null via API

### DISCUSSION THREAD

**Discussion by ghost** [pimcore/pimcore#10174] [Q&A / Support] [answered]
I have a lot of objects with differents classes. All my objects appear in Pimcore GUI and are published but for only some classes when I try to load them with ```getById``` or ```getByPath```, it returns Null...

**Comment by passioneight** [comment]
Here's what I'd do next:
1. Make sure both your ID and the path used to load the objects are correct by checking them in Pimcore.
    - Note that you should not hardcode these kind of things as they might vary for different environments.
2. If they are correct, try clearing the cache `bin/console cache:clear` & `bin/console pimcore:cache:clear`. If they are not correct, change your code accordingly.
3. If the first two steps didn't work, try the `$force` parameter.
4. Still nothing? Use `xdebug` to see what's going on and/or check your log files.

If none of this works for you, please provide more information, preferably steps to reproduce.

**Comment by ghost** [comment, marked as answer]
ok thank you for your advice @passioneight 
i found what do not work
in all of my path, i used brackets [] and it seems that it works for the first object but not for the following...
